src/transactions/transactionBuilder/transactionBuilder.ts

```diff
@@ -565,48 +537,52 @@ export function generateUserTransactionHash(args: InputSubmitTransactionData): s
 }
 
 /**
- * Derive the raw transaction type - FeePayerRawTransaction or MultiAgentRawTransaction or RawTransaction
+ * Generate a multi signers signed transaction that can be submitted to chain
  *
- * @param transaction A aptos transaction type
+ * @param transaction MultiAgentRawTransaction | FeePayerRawTransaction
+ * @param senderAuthenticator The account authenticator of the transaction sender
+ * @param secondarySignerAuthenticators The extra signers account Authenticators
  *
- * @returns FeePayerRawTransaction | MultiAgentRawTransaction | RawTransaction
+ * @returns A SignedTransaction
  */
-export function deriveTransactionType(transaction: AnyRawTransaction): AnyRawTransactionInstance {
-  if (transaction.feePayerAddress) {
-    return new FeePayerRawTransaction(
-      transaction.rawTransaction,
-      transaction.secondarySignerAddresses ?? [],
-      transaction.feePayerAddress,
+export function generateMultiSignersSignedTransaction(
+  transaction: MultiAgentRawTransaction | FeePayerRawTransaction,
+  senderAuthenticator: AccountAuthenticator,
+  feePayerAuthenticator?: AccountAuthenticator,
+  additionalSignersAuthenticators?: Array<AccountAuthenticator>,
+) {
+  if (transaction instanceof FeePayerRawTransaction) {
```

Comment:
`instanceOf` creates bundle issues, we switched to use type checking, check out https://github.com/aptos-labs/aptos-ts-sdk/pull/393/files